integrationtest/orm/src/test/java/org/hibernate/search/v6poc/integrationtest/orm/AbstractOrmAutomaticIndexingAssociationIT.java

```diff
@@ -0,0 +1,568 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.v6poc.integrationtest.orm;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.hibernate.SessionFactory;
+import org.hibernate.boot.Metadata;
+import org.hibernate.boot.MetadataSources;
+import org.hibernate.boot.SessionFactoryBuilder;
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.search.v6poc.entity.orm.cfg.SearchOrmSettings;
+import org.hibernate.search.v6poc.util.impl.integrationtest.common.rule.BackendMock;
+import org.hibernate.search.v6poc.util.impl.integrationtest.common.stub.backend.index.impl.StubBackendFactory;
+import org.hibernate.search.v6poc.util.impl.integrationtest.orm.OrmUtils;
+import org.hibernate.search.v6poc.util.impl.test.annotation.TestForIssue;
+import org.hibernate.service.ServiceRegistry;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+/**
+ * An abstract base for tests dealing with automatic indexing based on Hibernate ORM entity events
+ * and involving an association.
+ * <p>
+ * We use a contrived design based on a {@link AssociationModelPrimitives} class that defines all the factory methods,
+ * setters and getters we need,
+ * because we want to have separate model classes for every test,
+ * in order to avoid introducing exotic situations that would arise
+ * when using generics or superclasses in an ORM model.
+ * <p>
+ * Generally all models follow these guidelines:
+ * <ul>
+ *     <li>TIndexed is an @Indexed entity type</li>
+ *     <li>TContained is a non-@Indexed entity type</li>
+ *     <li>
+ *         TContaining is a non-abstract supertype of TIndexed which defines, in particular,
+ *         a one-to-one, @IndexedEmbedded, child association to TContaining,
+ *         and several @IndexedEmbedded associations (the ones under test) to TContained.
+ *     </li>
+ * </ul>
+ * <p>
+ * Most tests defined in this class use a very simple setup when it comes to associations,
+ * with one parent TIndexed linked to one child TContaining linked to one TContained.
+ * Subclasses define other tests with more advanced setups, with more assumptions as to what the associations allow.
+ */
+public abstract class AbstractOrmAutomaticIndexingAssociationIT<
+		TIndexed extends TContaining, TContaining, TContained
+		> {
+
+	private static final String PREFIX = SearchOrmSettings.PREFIX;
+
+	@Rule
+	public BackendMock backendMock = new BackendMock( "stubBackend" );
+
+	SessionFactory sessionFactory;
```

Comment:
Raaah, again and again ;). I think you really mean `protected` here as if I have a subclass of this one in a different package, you will expect your `sessionFactory` to be visible.

Reply:
I don't expect this class to be extended from a different package, that's why :) Anyway, it's just a test, so... I'll change it, just this once :p